client/src/hooks/store/useLeaderBoardStore.tsx

```diff
@@ -1,123 +1,234 @@
-import { getOrderName, orders } from "@bibliothecadao/eternum";
+import { getOrderName, HYPERSTRUCTURE_TOTAL_COSTS_SCALED, orders } from "@bibliothecadao/eternum";
 import { create } from "zustand";
 import { getEntityIdFromKeys } from "../../ui/utils/utils";
-import { getComponentValue } from "@dojoengine/recs";
+import { getComponentValue, HasValue } from "@dojoengine/recs";
 import { useDojo } from "../context/DojoContext";
 import { useGetRealms, useRealm } from "../helpers/useRealm";
-import { useEffect } from "react";
+import { useEffect, useRef, useState } from "react";
 import { useEntities } from "../helpers/useEntities";
 
-export interface PlayerResourceLeaderboardInterface {
+
+import { Subscription } from "rxjs";
+import useBlockchainStore from "../store/useBlockchainStore";
+import { useContributions } from "../helpers/useContributions"
+import { ResourcesIds } from "@bibliothecadao/eternum";
+
+// Define elsewhere
+const PTS_PER_CYCLE = 100;
+const CYCLE_TIME = 24 * 60 * 60 * 1000;
+
+// Computed based on qty of 10 of each resource for hyperstructure building
+const TOTAL_EFFECTIVE_CONTRIBUTION = 8209.10;
+
+// fix ?
+const ResourceMultipliers: { [key in ResourcesIds]?: number } = {
+  [ResourcesIds.Wood]: 1.00,
+  [ResourcesIds.Stone]: 1.27,
+  [ResourcesIds.Coal]: 1.31,
+  [ResourcesIds.Copper]: 1.90,
+  [ResourcesIds.Obsidian]: 2.26,
+  [ResourcesIds.Silver]: 2.88,
+  [ResourcesIds.Ironwood]: 4.25,
+  [ResourcesIds.ColdIron]: 5.24,
+  [ResourcesIds.Gold]: 5.49,
+  [ResourcesIds.Hartwood]: 8.44,
+  [ResourcesIds.Diamonds]: 16.72,
+  [ResourcesIds.Sapphire]: 20.30,
+  [ResourcesIds.Ruby]: 20.98,
+  [ResourcesIds.DeepCrystal]: 20.98,
+  [ResourcesIds.Ignium]: 29.15,
+  [ResourcesIds.EtherealSilica]: 30.95,
+  [ResourcesIds.TrueIce]: 36.06,
+  [ResourcesIds.TwilightQuartz]: 45.18,
+  [ResourcesIds.AlchemicalSilver]: 53.92,
+  [ResourcesIds.Adamantine]: 91.20,
+  [ResourcesIds.Mithral]: 135.53,
+  [ResourcesIds.Dragonhide]: 217.92,
+  [ResourcesIds.Earthenshard]: 20.98,
```

Comment:
this should be more, as it can only be found on the map. Change to 400.

Reply:
this is the same amount as in the contract currently (same as ruby). The weight will be 1/x of this though since this represents the rarity

Reply:
later on, this should be in config and it should be set in contract config as well rather than being constants in constants.cairo

Reply:
currently you can find them on the map + players who find a shards mine have infinite production of it, so i'm not sure it's necessarily more rare. Depends the prob of finding a shards mine